ratis-server/src/main/java/org/apache/ratis/server/storage/RaftStorageDirectoryImpl.java

```diff
@@ -150,6 +152,13 @@ StorageState analyzeStorage(boolean toLock) throws IOException {
       this.lock(); // lock storage if it exists
     }
 
+    // check enough space
+    if (!hasEnoughSpace()) {
+      LOG.warn("There are not enough space left for directory " + rootPath
+          + "required: " + spaceReserved + " left: " + root.getFreeSpace());
+      return StorageState.NON_EXISTENT;
```

Comment:
Let's add a new NO_SPACE state.  Otherwise, StartupOption.FORMAT won't work.

Reply:
Oh, thanks for pointing this out, will fix it.